mitodl/mitxonline
# Repair cart page including reinstated course api improvements

# What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/1876
https://github.com/mitodl/mitxonline/issues/1869

# Description (What does it do?)

1. Reinstates: https://github.com/mitodl/mitxonline/pull/1872 & https://github.com/mitodl/mitxonline/pull/1877
2. Fixes an issue when loading the cart page (https://mitxonline.mit.edu/cart/) with an item in the basket.  Issue ref: https://mit-office-of-digital-learning.sentry.io/issues/4472765075/?alert_rule_id=8724694&alert_type=issue&notification_uuid=2bbf21e8-fcf4-46ec-944a-9cc1c91f3c80&project=5864687&referrer=slack

# How can this be tested?

1. Enroll into a course and proceed with certificate enrollment.
2. When redirected to the cart page, ensure that the cart page loads as expected and there are no errors in the browser console.
3. Through http://mitxonline.odl.local:8013/cms/, add a flexible price child page to the course page for the course you're enrolled in.
4. Edit the course page and add the URL of the flexible price page to the URL field under the "price details" section.
5. Refresh the cart page and ensure that you see a "Need financial assistance?" link.

# Additional Context
This was caused because I was passing a course to `isFinancialAssistanceAvailable`, here: https://github.com/mitodl/mitxonline/blob/c87bdc15e75c6bae728749dedd5a91cae3755c8d/frontend/public/src/containers/pages/checkout/CartPage.js#L141C15-L141C15
